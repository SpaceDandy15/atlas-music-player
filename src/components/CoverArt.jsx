import React from "react";
import placeholder from "../assets/placeholder.svg"; // fallback if no src provided

export default function CoverArt({ src }) {
  return (
    <div className="w-48 h-48 overflow-hidden rounded-lg">
      <img
        src={src || placeholder}
        alt="Cover Art"
        className="w-full h-full object-cover"
      />
    </div>
  );
}
