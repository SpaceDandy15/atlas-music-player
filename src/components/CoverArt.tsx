import React from "react";
import placeholder from "../assets/placeholder.svg";

type CoverArtProps = {
  src?: string; // falls back to placeholder
};

export default function CoverArt({ src }: CoverArtProps) {
  return (
    <div className="w-48 h-48 overflow-hidden rounded-lg">
      <img
        src={src || placeholder}
        alt="Cover Art"
        className="w-full h-full object-cover"
      />
    </div>
  );
}
