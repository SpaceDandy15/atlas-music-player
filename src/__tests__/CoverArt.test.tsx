import React from "react";
import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import CoverArt from "../components/CoverArt";
import renderer from "react-test-renderer";

describe("CoverArt Component", () => {
  it("renders correctly with a provided src", () => {
    // Snapshot test for when a specific cover image URL is passed
    const tree = renderer
      .create(<CoverArt src="https://example.com/cover.jpg" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly without a src (uses placeholder)", () => {
    // Snapshot test for fallback scenario when no src is provided
    const tree = renderer.create(<CoverArt />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
