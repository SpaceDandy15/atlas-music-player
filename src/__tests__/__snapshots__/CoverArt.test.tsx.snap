// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt Component > renders correctly with a provided src 1`] = `null`;

exports[`CoverArt Component > renders correctly without a src (uses placeholder) 1`] = `null`;
